Automattic/jetpack
# CRM: Add automation workflows table

This PR introduces a new automations workflow table.

It could be argued that the table should be named "automations" but that would IMO not be the correct name since a workflow is "a named datatype" in the overall automations systems and when someone wanted to look up a workflow, they would not look for the "automations" table.

An additional benefit is that it can co-exist next to the current automations table as well, so we're not stepping on any toes while hiding the new automation system behind a feature flag. 

Fixes https://github.com/Automattic/zero-bs-crm/issues/3198

**Update**

I've removed `zbs_team` as a column even though it was suggested by @cleacos.
The main reason being that I don't see us going to that level of customization any time soon where specific teams can set up their own workflows + "automations v1" didn't include it either.

Honestly, I contemplated not including the owner as well, but seeing who created the workflow is probably OK - but we're basically using it as "created by" instead of an "owner" at that point.

## Proposed changes:

* Add a new `workflows` table.
* Add a migration script that ensures the workflow table is added.

### Other information:


## Jetpack product discussion



* p1693381526547099/1693338179.961359-slack-CTXBP902X
* pbhBOz-2ES-p2
* pbhBOz-3Wf-p2

## Does this pull request change what data or activity we track or use?



No.

## Testing instructions:






* Spin up a CRM installation using this patch
* Load any page (e.g. go to `/wp-admin`)
* Check that the new `*_zbs_workflows` table exists
* Check that the structure of the table matches what is described in the code
